In [1]:
import pandas as pd

**1. Introduction**

In [20]:
'''
Main analysis
'''

df = pd.read_csv('Amazon_Sale_Report.csv', index_col='index')
print(f"Rows, columns: {df.shape} ")
df.head(n=3)

Rows, columns: (128975, 23) 


/tmp/ipython-input-383/1608414725.py:5: DtypeWarning: Columns (23) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('Amazon_Sale_Report.csv', index_col='index')


Order ID      Date                        Status Fulfilment  \
index                                                                           
0      405-8078784-5731545  04-30-22                     Cancelled   Merchant   
1      171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   Merchant   
2      404-0687676-7273146  04-30-22                       Shipped     Amazon   

      Sales Channel  ship-service-level    Style              SKU Category  \
index                                                                        
0          Amazon.in           Standard   SET389   SET389-KR-NP-S      Set   
1          Amazon.in           Standard  JNE3781  JNE3781-KR-XXXL    kurta   
2          Amazon.in          Expedited  JNE3371    JNE3371-KR-XL    kurta   

      Size  ... currency  Amount    ship-city   ship-state  ship-postal-code  \
index       ...                                                                
0        S  ...      INR  647.62       MUMBAI  MAHARASHTRA          400081.0   
1      3XL  ...      INR  406.00    BENGALURU    KARNATAKA          560085.0   
2       XL  ...      INR  329.00  NAVI MUMBAI  MAHARASHTRA          410210.0   

      ship-country                                      promotion-ids    B2B  \
index                                                                          
0               IN                                                NaN  False   
1               IN  Amazon PLCC Free-Financing Universal Merchant ...  False   
2               IN       IN Core Free Shipping 2015/04/08 23-48-5-108   True   

      fulfilled-by Unnamed: 22  
index                           
0        Easy Ship         NaN  
1        Easy Ship         NaN  
2              NaN         NaN  

[3 rows x 23 columns]

**2. Data Overview**

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 128975 entries, 0 to 128974
Data columns (total 23 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Order ID            128975 non-null  object 
 1   Date                128975 non-null  object 
 2   Status              128975 non-null  object 
 3   Fulfilment          128975 non-null  object 
 4   Sales Channel       128975 non-null  object 
 5   ship-service-level  128975 non-null  object 
 6   Style               128975 non-null  object 
 7   SKU                 128975 non-null  object 
 8   Category            128975 non-null  object 
 9   Size                128975 non-null  object 
 10  ASIN                128975 non-null  object 
 11  Courier Status      122103 non-null  object 
 12  Qty                 128975 non-null  int64  
 13  currency            121180 non-null  object 
 14  Amount              121180 non-null  float64
 15  ship-city           128942 non-null  ob

In [5]:
df.describe()

Qty         Amount  ship-postal-code
count  128975.000000  121180.000000     128942.000000
mean        0.904431     648.561465     463966.236509
std         0.313354     281.211687     191476.764941
min         0.000000       0.000000     110001.000000
25%         1.000000     449.000000     382421.000000
50%         1.000000     605.000000     500033.000000
75%         1.000000     788.000000     600024.000000
max        15.000000    5584.000000     989898.000000

As we can see, dataset contains 128975 rows and 23 columns. Several columns have missing values - for example column 'Unnamed' is very suspicious. Let's make a copy of our dataset to ensure that original data will stay untouched.

In [9]:
df_copy = df.copy()

**3. Missing values analysis**

The first column to further examination is 'Unnamed' column.

In [10]:
df_copy['Unnamed: 22']

index
0           NaN
1           NaN
2           NaN
3           NaN
4           NaN
          ...  
128970    False
128971    False
128972    False
128973    False
128974    False
Name: Unnamed: 22, Length: 128975, dtype: object

In [19]:
unnamed_counts = df_copy['Unnamed: 22'].value_counts()
print(f"Values in 'Unnamed 22' column: {unnamed_counts}")
print("")
unnamed_missing = df_copy['Unnamed: 22'].isna().sum()
print(f"Values missing in column 'Unnamed 22': {unnamed_missing}")

Values in 'Unnamed 22' column: Unnamed: 22
False    79925
Name: count, dtype: int64

Values missing in column 'Unnamed 22': 49050


As we can see - this column is not truly informative.  <br>
✅ - TO DO: Column 'Unnamed 22' is to be deleted.

In [24]:
df_copy.drop('Unnamed: 22', axis=1, inplace=True)
df_copy.head(3)

Order ID      Date                        Status Fulfilment  \
index                                                                           
0      405-8078784-5731545  04-30-22                     Cancelled   Merchant   
1      171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   Merchant   
2      404-0687676-7273146  04-30-22                       Shipped     Amazon   

      Sales Channel  ship-service-level    Style              SKU Category  \
index                                                                        
0          Amazon.in           Standard   SET389   SET389-KR-NP-S      Set   
1          Amazon.in           Standard  JNE3781  JNE3781-KR-XXXL    kurta   
2          Amazon.in          Expedited  JNE3371    JNE3371-KR-XL    kurta   

      Size  ... Qty currency  Amount    ship-city   ship-state  \
index       ...                                                  
0        S  ...   0      INR  647.62       MUMBAI  MAHARASHTRA   
1      3XL  ...   1      INR  406.00    BENGALURU    KARNATAKA   
2       XL  ...   1      INR  329.00  NAVI MUMBAI  MAHARASHTRA   

      ship-postal-code ship-country  \
index                                 
0             400081.0           IN   
1             560085.0           IN   
2             410210.0           IN   

                                           promotion-ids    B2B fulfilled-by  
index                                                                         
0                                                    NaN  False    Easy Ship  
1      Amazon PLCC Free-Financing Universal Merchant ...  False    Easy Ship  
2           IN Core Free Shipping 2015/04/08 23-48-5-108   True          NaN  

[3 rows x 22 columns]

Let's check other missing values.

In [27]:
df_copy.isnull().sum().sort_values()

Order ID                  0
Date                      0
Status                    0
Fulfilment                0
Sales Channel             0
ship-service-level        0
Style                     0
SKU                       0
Category                  0
Size                      0
ASIN                      0
Qty                       0
B2B                       0
ship-city                33
ship-state               33
ship-country             33
ship-postal-code         33
Courier Status         6872
Amount                 7795
currency               7795
promotion-ids         49153
fulfilled-by          89698
dtype: int64

As we can see - other columns to further examination are: fulfilled-by, promotion-ids, currency, Amount. But first - let's change all column names to lower case. <br>
✅ - TO DO: change column names to lower

In [28]:
df_copy.columns = df_copy.columns.str.lower()


order id      date                        status  \
index                                                                 
0       405-8078784-5731545  04-30-22                     Cancelled   
1       171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   
2       404-0687676-7273146  04-30-22                       Shipped   
3       403-9615377-8133951  04-30-22                     Cancelled   
4       407-1069790-7240320  04-30-22                       Shipped   
...                     ...       ...                           ...   
128970  406-6001380-7673107  05-31-22                       Shipped   
128971  402-9551604-7544318  05-31-22                       Shipped   
128972  407-9547469-3152358  05-31-22                       Shipped   
128973  402-6184140-0545956  05-31-22                       Shipped   
128974  408-7436540-8728312  05-31-22                       Shipped   

       fulfilment sales channel  ship-service-level    style              sku  \
index                                                                           
0        Merchant      Amazon.in           Standard   SET389   SET389-KR-NP-S   
1        Merchant      Amazon.in           Standard  JNE3781  JNE3781-KR-XXXL   
2          Amazon      Amazon.in          Expedited  JNE3371    JNE3371-KR-XL   
3        Merchant      Amazon.in           Standard    J0341       J0341-DR-L   
4          Amazon      Amazon.in          Expedited  JNE3671  JNE3671-TU-XXXL   
...           ...            ...                ...      ...              ...   
128970     Amazon      Amazon.in          Expedited  JNE3697    JNE3697-KR-XL   
128971     Amazon      Amazon.in          Expedited   SET401   SET401-KR-NP-M   
128972     Amazon      Amazon.in          Expedited    J0157     J0157-DR-XXL   
128973     Amazon      Amazon.in          Expedited    J0012     J0012-SKD-XS   
128974     Amazon      Amazon.in          Expedited    J0003      J0003-SET-S   

             category size  ... qty currency   amount    ship-city  \
index                       ...                                      
0                 Set    S  ...   0      INR   647.62       MUMBAI   
1               kurta  3XL  ...   1      INR   406.00    BENGALURU   
2               kurta   XL  ...   1      INR   329.00  NAVI MUMBAI   
3       Western Dress    L  ...   0      INR   753.33   PUDUCHERRY   
4                 Top  3XL  ...   1      INR   574.00      CHENNAI   
...               ...  ...  ...  ..      ...      ...          ...   
128970          kurta   XL  ...   1      INR   517.00    HYDERABAD   
128971            Set    M  ...   1      INR   999.00     GURUGRAM   
128972  Western Dress  XXL  ...   1      INR   690.00    HYDERABAD   
128973            Set   XS  ...   1      INR  1199.00        Halol   
128974            Set    S  ...   1      INR   696.00       Raipur   

          ship-state ship-postal-code ship-country  \
index                                                
0        MAHARASHTRA         400081.0           IN   
1          KARNATAKA         560085.0           IN   
2        MAHARASHTRA         410210.0           IN   
3         PUDUCHERRY         605008.0           IN   
4         TAMIL NADU         600073.0           IN   
...              ...              ...          ...   
128970     TELANGANA         500013.0           IN   
128971       HARYANA         122004.0           IN   
128972     TELANGANA         500049.0           IN   
128973       Gujarat         389350.0           IN   
128974  CHHATTISGARH         492014.0           IN   

                                            promotion-ids    b2b fulfilled-by  
index                                                                          
0                                                     NaN  False    Easy Ship  
1       Amazon PLCC Free-Financing Universal Merchant ...  False    Easy Ship  
2            IN Core Free Shipping 2015/04/08 23-48-5-108   True          NaN  
3                          